app/services/onboarding.ts

```diff
@@ -172,56 +190,114 @@ class OnboardingViews extends ViewHandler<IOnboardingServiceState> {
   }
 
   get steps() {
-    const steps: IOnboardingStep[] = [];
     const userViews = this.getServiceViews(UserService);
     const isOBSinstalled = this.getServiceViews(ObsImporterService).isOBSinstalled();
     const recordingModeEnabled = this.getServiceViews(RecordingModeService).isRecordingModeEnabled;
 
-    if (process.platform === OS.Mac) {
-      steps.push(ONBOARDING_STEPS()[EOnboardingSteps.MacPermissions]);
-    }
-
-    steps.push(ONBOARDING_STEPS()[EOnboardingSteps.SteamingOrRecording]);
+    const streamerKnowledgeMode = this.streamerKnowledgeMode;
+
+    const { existingSceneCollections, importedFromObs } = this.state;
+    const { isLoggedIn, isPrime: isUltra } = userViews;
+
+    const ctx: OnboardingStepContext = {
+      streamerKnowledgeMode,
+      recordingModeEnabled,
+      existingSceneCollections,
+      importedFromObs,
+      isLoggedIn,
+      isUltra,
+      isObsInstalled: isOBSinstalled,
+      isPartialSLAuth: userViews.auth && userViews.isPartialSLAuth,
+      platformSupportsThemes:
+        isLoggedIn && getPlatformService(userViews.platform?.type)?.hasCapability('themes'),
+    };
 
-    steps.push(ONBOARDING_STEPS()[EOnboardingSteps.Connect]);
+    return this.getStepsForMode(streamerKnowledgeMode)(ctx);
+  }
 
-    if (userViews.auth && userViews.isPartialSLAuth) {
-      steps.push(ONBOARDING_STEPS()[EOnboardingSteps.PrimaryPlatformSelect]);
-    }
+  get totalSteps() {
+    return this.steps.length;
+  }
 
-    if (isOBSinstalled && !recordingModeEnabled) {
-      steps.push(ONBOARDING_STEPS()[EOnboardingSteps.FreshOrImport]);
+  getStepsForMode(mode: StreamerKnowledgeMode) {
+    const { getSteps } = this;
+
+    switch (mode) {
+      case StreamerKnowledgeMode.BEGINNER:
+        return getSteps([
+          EOnboardingSteps.MacPermissions,
+          EOnboardingSteps.StreamingOrRecording,
+          EOnboardingSteps.Connect,
+          EOnboardingSteps.PrimaryPlatformSelect,
+          EOnboardingSteps.FreshOrImport,
+          EOnboardingSteps.ObsImport,
+          EOnboardingSteps.HardwareSetup,
+          EOnboardingSteps.ThemeSelector,
+          EOnboardingSteps.Prime,
+          EOnboardingSteps.Tips,
+        ]);
+      case StreamerKnowledgeMode.INTERMEDIATE:
+        /*
+         * Yes, these are the same as beginner, only inner screens are supposed to differ,
+         * but the one screen that was provided is currently disabled (Optimizer).
+         * Nevertheless, this sets the foundation for future changes.
+         */
+        return getSteps([
+          EOnboardingSteps.MacPermissions,
+          EOnboardingSteps.StreamingOrRecording,
+          EOnboardingSteps.Connect,
+          EOnboardingSteps.PrimaryPlatformSelect,
+          EOnboardingSteps.FreshOrImport,
+          EOnboardingSteps.ObsImport,
+          EOnboardingSteps.HardwareSetup,
+          EOnboardingSteps.ThemeSelector,
+          EOnboardingSteps.Prime,
+          EOnboardingSteps.Tips,
+        ]);
```

Comment:
just kind of a fun thing we can do
```
case StreamerKnowlegeMode.BEGINNER:
case StreamerKnowledgeMode.INTERMEDIATE:
  return getSteps(...)
```

if they are the same but if we are planning for them to diverge in the future i get not doing it too

Reply:
Same idea, I thought to keep them separate to make it explicit, the next time we add a step we would add it to a list instead of copying all over from the other mode first